portal/src/main/webapp/js/src/over-representation-analysis/view.js

```diff
@@ -253,6 +284,86 @@ var orTable = function() {
 
     }
 
+    /**
+     * when a gene in the table is clicked, show the gene in the mini-onco
+     */
+    function addGeneClick(){
+        $('.selectHighlight_'+table_id).on('click', function() {
+            var current_gene = $(this).text();
+            self.miniOnco.render(current_gene);
+        });
+    }
+
+    /**
+     * show the loading icon and decrease visibility of table
+     */
+    function startLoading() {
+        $("#" + table_id).addClass('tableLoading');
+        $('#' + loading_div).addClass('loaderIconLoading');
+    }
+
+    /**
+     * hide the loading icon and restore the visibility of table
+     */
+    function stopLoading() {
+        $("#" + table_id).removeClass('tableLoading');
+        $('#' + loading_div).removeClass('loaderIconLoading');
+    }
+
+    /**
+     * Search the table by building a regular expression
+     * Shows the loading icon while searching
+     *
+     * @param selection: list of genes to search for
+     */
+    this.searchTable = function(selection) {
+        // set loading icon for table
+        startLoading();
+
+        // add timeout to actually show the loading image
+        setTimeout(function() {
+            var searchExpression = "";
+
+            // if no genes, search will be with an empty filter, otherwise create a (maybe extremely long...) searchExpression
+            if (selection.length > 0) {
+                searchExpression = "^" + selection.join("$|^") + "$";
+            }
+            //apply search expression to the dataTable:
+            orTableInstance.DataTable().column(0).search(
+                searchExpression,
+                true,
+                false
+            ).draw();
+
+            // stop loading
+            stopLoading();
+        }, 1);
+    }
+
+    /**
+     * check whether the profile_type supports a mini-onco. This is the case for mutations and copy number
+     * @returns {boolean}
+     */
+    this.supportsMiniOnco = function(){
+        return profile_type === orAnalysis.profile_type.mutations || profile_type === orAnalysis.profile_type.copy_num;
+    }
+
+    /**
+     * check whether the datatype is LOG-VALUE to prevent logging it again
+     * @returns {boolean}
+     */
+    this.hasLogData = function(){
+        return data_type === "LOG-VALUE";
+    }
```

Comment:
@jjgao : be aware this could be a pitfall! You need to check if all the relevant genetic profiles on your server have the data type correctly set to LOG-VALUE, otherwise they will not be selected here!